lib/src/widgets/master_detail/yaru_master_detail_page.dart

```diff
@@ -163,29 +167,31 @@ class _YaruMasterDetailPageState extends State<YaruMasterDetailPage> {
 
   @override
   Widget build(BuildContext context) {
-    return Scaffold(
-      body: _YaruMasterDetailLayoutBuilder(
-        portrait: (context) => YaruPortraitLayout(
-          tileBuilder: widget.tileBuilder,
-          pageBuilder: widget.pageBuilder,
-          onSelected: widget.onSelected,
-          appBar: widget.appBar ?? widget.appBarBuilder?.call(context),
-          bottomBar: widget.bottomBar,
-          controller: _controller,
-        ),
-        landscape: (context) => YaruLandscapeLayout(
-          tileBuilder: widget.tileBuilder,
-          pageBuilder: widget.pageBuilder,
-          onSelected: widget.onSelected,
-          layoutDelegate: widget.layoutDelegate,
-          previousPaneWidth: _previousPaneWidth,
-          onLeftPaneWidthChange: (width) => _previousPaneWidth = width,
-          appBar: widget.appBar ?? widget.appBarBuilder?.call(context),
-          bottomBar: widget.bottomBar,
-          controller: _controller,
-        ),
-      ),
-    );
+    return widget.length == 0 || widget.controller?.length == 0
+        ? widget.emptyBuilder?.call(context) ?? const SizedBox.shrink()
```

Comment:
what do you think about moving this inside the scaffold?

i'm thinking about the kind of structure where you have `YaruMasterDetailPage` right below `MaterialApp`. if we have the empty builder inside the `Scaffold`, we wouldn't get errors about missing `Material` etc.

Reply:
Good point. How can we handle this in the `YaruNavigationPage`? We have `LayoutBuilder`->`SafeArea`->`Scaffold` there. Is the `SafeArea` needed for anything, and can we just move the `Scaffold` to the top? (On first glance it seems to work, but maybe there are reasons for this widget hierarchy I'm not aware of)

Reply:
Not sure where SafeArea came from and even less sure if it does anything on desktop platforms...